### ISSUE TITLE
SharedStateServer doesn't send update to client after refresh

### ISSUE THREAD

**Issue by brentbaum** [idom-team/idom#352]
I have a simple app with a few buttons. When the user clicks on a button, the text next to the button toggles between "asserted=False" and "asserted=True".

<img width="388" alt="Screen Shot 2021-04-22 at 8 51 47 AM" src="https://user-images.githubusercontent.com/868955/115746199-22267400-a362-11eb-83b2-52f3245fd68b.png">

The first time a user brings up the app everything works as expected. The network tab looks like this.

<img width="525" alt="Screen Shot 2021-04-22 at 8 51 51 AM" src="https://user-images.githubusercontent.com/868955/115746669-821d1a80-a362-11eb-93d0-833382b44ecc.png">

After a refresh the app keeps sending events but stops receiving them.

<img width="546" alt="Screen Shot 2021-04-22 at 8 52 07 AM" src="https://user-images.githubusercontent.com/868955/115746706-8a755580-a362-11eb-8a3b-0a9b92a19e24.png">

This behavior is consistent across any components that depend on state. Notably, the user is unable to change the value of inputs.

Server code:

```
def start_app(dispatch):
    from idom.server.sanic import SharedClientStateServer

    # we use a redux-like dispatch structure to send events to outside the idom event, then update state
    # with a callback stored outside the app. 
    # I've tried removing this particular piece of functionality and the problem persists.
    root = lambda: Panel(dispatch)
    app = SharedClientStateServer(root)
    return app.daemon("0.0.0.0", 3040)
```

Any ideas on the root cause or a way to mitigate? The PerClientStateServer throws a CannotAccessError due to our rapid state updates, logging `(f"Did not update {component} - resources of {self} are closed")` so it's a no-go (sorry, I've been waiting to file an issue on that one!)

Thanks!

**Comment by rmorshea** [comment]
Did this issue arise after the latest update?

**Comment by brentbaum** [comment]
@rmorshea I've had this issue since 0.22 and haven't tested before that. 

You got it, the `CannotAccessError` (really, just the log message but that's triggered by a `CannotAccessError`) happens after refresh.

**Comment by rmorshea** [comment]
Ok, it looks like I'm able to reproduce, so I'll look into that.

With that said, ignoring the logged `CannotAccessError`, does `PerClientStateServer` work for your purposes? Perhaps you could leverage a `use_effect` hook to avoid attempting to update the layout after it's been closed due to a refresh? It might be similar to [this example](https://idom-docs.herokuapp.com/docs/life-cycle-hooks.html#cleaning-up-effects) except instead of a connection, you'd be opening and closing your async thread of execution. For example:

```python
@idom.component
def SyncStateComponent():
    state, set_state = idom.hooks.use_state(None)

    @idom.hooks.use_effect
    def sync_state():
        stop_updating = start_updating(set_state)
        return stop_updating

    return ...

def start_updating(set_state):
    """Start async thread of execution to frequently set state and return callback to stop""" 
    ...
    return callback_to_stop_setting_state
```

The `SharedClientStateServer` is designed to give all clients the same synchronized view.

**Comment by rmorshea** [comment]
The main purpose of the `SharedClientStateServer` is that it allows all clients to share one synchronized view. For example, a caused or seen by one client will seen by all other clients too. Is that what you're trying to do?

Unfortunately that information isn't really documented yet.

**Comment by brentbaum** [comment]
One synchronized view is exactly what we're looking for; the SharedClientStateServer is working fantastically minus the refresh bug. I felt like it was pretty clear what the intent was based on the name :) I was falling back to the `PerClientStateServer` to allow refreshing (helpful during development).

The `PerClientStateServer` works as expected minus the log messages. I'm depending on logging heavily at this point so the frequent messages were a big enough annoyance that I moved away from that approach.

I've had difficulty with use_effect cleanup effects not running on window close / refresh before. I don't have anything reproducible at this point but will submit another issue if it becomes a problem.

**Comment by rmorshea** [comment]
I think I found the problem. I think it's related to this issue: https://github.com/idom-team/idom/issues/298

**Comment by rmorshea** [comment]
Ok, got the PR with the fix passing. Will try and release some time this week.

**Comment by rmorshea** [comment]
Fixed by #353
